### ISSUE TITLE
False positive with collections of type literals

### ISSUE THREAD

**Issue by ipkiss42** [microsoft/pyright#1901]
**Describe the bug**
When a type is defined as a literal, collections of that type seem to "lose" the literal information, causing false positives later on.

**To Reproduce**

With this code:
```python
from typing import Iterable, Set
from typing_extensions import Literal

FooOrBar = Literal["foo", "bar"]

def to_set(values: Iterable[FooOrBar]) -> Set[FooOrBar]:
    return {value for value in values}
```

Pyright wrongly complains about typing:

```
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/pyright1/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/pyright1/foo.py
  /tmp/pyright1/foo.py:7:12 - error: Expression of type "set[str]" cannot be assigned to return type "Set[FooOrBar]"
    TypeVar "_T@set" is invariant
      Type "str" cannot be assigned to type "FooOrBar"
        "str" cannot be assigned to type "Literal['foo']" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
Completed in 0.645sec
```

**Expected behavior**
No issue reported

**VS Code extension or command-line**
Command-line (Pyright 1.1.142)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.143, which I just published. It will also be included in the next release of pylance.
